src/main/java/com/iota/iri/network/impl/TransactionRequesterWorkerImpl.java

```diff
@@ -109,7 +109,8 @@ public void processRequestQueue() {
         try {
             if (transactionRequester.numberOfTransactionsToRequest() >= REQUESTER_THREAD_ACTIVATION_THRESHOLD) {
                 TransactionViewModel transaction = getTransactionToSendWithRequest();
-                if (transaction != null && transaction.getType() != TransactionViewModel.PREFILLED_SLOT) {
+                if (transaction != null && (transaction.getType() != TransactionViewModel.PREFILLED_SLOT
+                        || transaction.getHash().equals(Hash.NULL_HASH))) {
```

Comment:
Note that usually we would do `Hash.Null_HASH.equals(transaction.getHash())` to avoid null pointer exceptions.

However, `transaction.getHash()` shouldn't be null and we don't want to be masking bugs. So this is actually good :-)

Reply:
Totally intended! :angel: !